### PR TITLE
Infra as a code by own your zinst packages

### PR DESCRIPTION
**Pull request by goody80**
For the centralized package manage & distributed systems
Package install manager. It very similar that concept of yinst command in Yahoo!

### CODE REVIEW

File: templates/zinst/0/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+version: '2'
+services:
+  zinst-repository:
+    image: zinst/zinst_repository:0.4
+    ports:
+    - 8080:80/tcp
+    volumes:
+    - /data/dist
```
Comment (rawmind0) [inline templates/zinst/0/docker-compose.yml]: Add volume definition
```
-    - /data/dist
+    - zinst-data:/data/dist
+
+volumes:
+  zinst-data:
+    driver: local
```

File: templates/zinst/0/rancher-compose.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      description: "public port to access the zinst repository"
+      label: "Public Port"
+      required: true
+      default: "8080"
+      type: "int"
+
+zinst-repository:
+  retain_ip: true
+  scale: 1
+  start_on_create: true
```
Comment (rawmind0) [inline templates/zinst/0/rancher-compose.yml]: Compose version 2 format, healthcheck and lb
```
-zinst-repository:
-  retain_ip: true
-  scale: 1
-  start_on_create: true
+services:
+  zinst-repository:
+    retain_ip: true
+    scale: 1
+    start_on_create: true
+    health_check:
+      port: 80
+      interval: 5000
+      unhealthy_threshold: 3
+      request_line: ''
+      healthy_threshold: 2
+      response_timeout: 5000
+  zinst-lb:
+    scale: 1
+    start_on_create: true
+    lb_config:
+      certs: []
+      port_rules:
+      - priority: 1
+        protocol: http
+        service: zinst-repository
+        source_port: ${public_port}
+        target_port: 80
+    health_check:
+      response_timeout: 2000
+      healthy_threshold: 2
+      port: 42
+      unhealthy_threshold: 3
+      interval: 2000
```

File: templates/zinst/0/rancher-compose.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+.catalog:
```
Comment (rawmind0) [inline templates/zinst/0/rancher-compose.yml]: Compose version 2 format.
```
-.catalog:
+version: '2'
+catalog:
```

File: templates/zinst/0/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+version: '2'
+services:
```
Comment (rawmind0) [inline templates/zinst/0/docker-compose.yml]: Publish ports through lb instead from the service.
```
+  zinst-lb:
+    image: rancher/lb-service-haproxy:v0.6.4
+    ports:
+    - ${public_port}:${public_port}/tcp
```

File: templates/zinst/0/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+  zinst-repository:
+    image: zinst/zinst_repository:0.4
+    ports:
+    - 8080:80/tcp
```
Comment (rawmind0) [inline templates/zinst/0/docker-compose.yml]: Port it's not published with ${public_port} value.
Anyway, better to use lb to publish ports, don't needed. 
```
-    ports:
-    - 8080:80/tcp
```

**Comment by rawmind0** [comment]
Hi @goody80 ... thanks so much for the PR and contribute... :)

Please, take a look at the change request.

**Comment by goody80** [comment]
Oh! Thanks! :)

File: templates/zinst/0/docker-compose.yml (hunk 4 above)
Comment (goody80) [inline templates/zinst/0/docker-compose.yml]: good! thanks :)

File: templates/zinst/0/docker-compose.yml (hunk 5 above)
Comment (goody80) [inline templates/zinst/0/docker-compose.yml]: thanks :)

**Comment by goody80** [comment]
How can I fix that ?

=================================== FAILURES ===================================
106s
335
________________________________ test_versions _________________________________
106s
337
templates = {'resourceType': u'template', 'data': [{'upgradeVersionLinks': None, 'path': u'library/mesos*zookeeper', 'bindings': N...cherVersion': u'', 'files': None, 'license': u'', 'templateBase': u'', 'isSystem': u'', 'description': u'Git Server'}]}
106s
339
    def test_versions(templates):
106s
340
        for template in templates:
106s
341
            # default version must be defined
106s
342
            assert template.defaultVersion
106s
343
            # template with default version must be defined
106s
344
>           assert template.versionLinks[template.defaultVersion]
106s
345
E           KeyError: u'v7.0.5'
106s
347
test_catalog.py:109: KeyError
106s
348
===================== 1 failed, 4 passed in 31.77 seconds ======================
106s
349
ERROR: InvocationError: '/scratch/integration/.tox/py27/bin/py.test --duration=20'
106s
350
___________________________________ summary ____________________________________
106s
351
  flake8: commands succeeded
106s
352
ERROR:   py27: commands failed

**Comment by rawmind0** [comment]
Catalog version at templates/zinst/0/rancher-compose.yml, ```7.0.5``` must match with version at templates/zinst/config.yml, ```v7.0.5``` ....

**Comment by goody80** [comment]
Oh! Thanks! I just changed :)

**Comment by rawmind0** [comment]
Hi @goody80 ...

I've deployed the package but i get this errors on zinst container
```
08/11/2017 13:26:40/bin/sh: -c: line 0: unexpected EOF while looking for matching `"'
08/11/2017 13:26:40/bin/sh: -c: line 1: syntax error: unexpected end of file
```

Could you please take a look??

**Comment by goody80** [comment]
@rawmind0 
Oh! okay let me check :)

**Comment by goody80** [comment]
@rawmind0 
I just updated the docker container for fix that error log. 

Thanks :)

**Comment by rawmind0** [comment]
Tested and working.... 

LGTM...

**Pull request merged**